src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_ALPHABETIACALLY = 'sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
+
+function visibleListGoods(
+  goods: string[],
+  sortFild: string,
+  reverseGood: boolean,
+) {
+  let sortGoods: string[] = [...goods];
+
+  if (sortFild) {
+    sortGoods = sortGoods.sort((good1, good2) => {
+      switch (sortFild) {
+        case SORT_ALPHABETIACALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseGood) {
+    sortGoods.reverse();
+  }
+
+  return sortGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortFild, setSortFild] = useState('');
+  const [reverseGood, setReverseGood] = useState(false);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortFild !== SORT_ALPHABETIACALLY,
+          })}
+          onClick={() => {
+            setSortFild(SORT_ALPHABETIACALLY);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortFild !== SORT_BY_LENGTH,
+          })}
+          onClick={() => {
+            setSortFild(SORT_BY_LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': reverseGood !== true,
+          })}
+          onClick={() => {
+            setReverseGood(!reverseGood);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortFild !== '' && (
```

Comment:
show the reset button in this case too
<img width="729" alt="image" src="https://github.com/mate-academy/react_list-of-goods-hooks/assets/70220667/4d1686e4-730e-44d9-b8de-563af0dbe9c3">